### PR TITLE
Wizard: validate wizard with Redux Hook

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Use redux hook to validate the form.
This gives us single point of contact for "is the data valid?" while not requiring every redux action touching form data to perform validation.

It's not perfect and might be improved when using external library solving the problems we're having.

Imperfections:
* Pristine state needs to be handled manually
* maintaining the validations externally to the fields

Possible future libraries (reserach needed):
* Redux form - deprecated, but that's what we do currently
* [Final form](https://final-form.org/) considered one of the best, but mostly based on data not being in redux
  * IMO best fit and we should embrace not having our form data in redux :) 
* [React hook form](https://www.react-hook-form.com/) (uses mostly uncontrolled componente - not great fit

### CODE REVIEW

**Review by regexowl — COMMENTED** [review]
Thanks for looking into validation! :sparkles: Added some comments.

As to what we could possibly use to make our situation easier - we discussed that quite extensively when the migration to V2 started and settled on using Redux state as we already used Redux and it seemed like the best option. It was months ago so some inaccuracies might follow :sweat_smile:

The point of migrating from V1 to V2 was getting rid of Data Driven Forms which are based on React Final Form as the form manager. And the problem with (React) Final Form is that it's no longer maintained. There might have been some dep bumps 10 months ago, but most of the code wasn't touched for years. There were a lot of troubles and blockers with DDF / Final Form and it started becoming technical debt.

React Hook Form is its modern and maintained alternative, but as you've mentioned it's not a great fit. Mainly because it's not compatible with PatternFly components, which would require a lot of workarounds and tinkering. There was an attempt to make it work, but it was unsuccessful.

We don't use Redux form, but some form of very manual (but also quite controlled) data management. I'll admit it's not super ideal and I'm open for more suggestions.

But also yeah as mentioned - storing everything "manually" in the Redux store was a very conscious choice. :melting_face:

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (5.0.5) is older than required version (6.0.0)

File: src/Components/CreateImageWizardV2/steps/FileSystem/FileSystemConfiguration.tsx [hunk 1]
```diff
@@ -193,12 +143,10 @@ export const Row = ({
   onDrop,
 }: RowPropTypes) => {
   const dispatch = useAppDispatch();
-  const partitions = useAppSelector(selectPartitions);
   const handleRemovePartition = (id: string) => {
     dispatch(removePartition(id));
   };
-  const isNextButtonPristine = useAppSelector(selectIsNextButtonTouched);
-  const duplicates = getDuplicateMountPoints(partitions);
+  const stepValidation = useFilesystemValidation();
```
[the remark this answers is not in the document]
Comment (ezr-ondrej) [reply src/Components/CreateImageWizardV2/steps/FileSystem/FileSystemConfiguration.tsx:150]: It's not amazing, but I've managed to simulate the same behaviour :)
See `beforeNext` in the CustomFooter with Context to propagate the pristine info to the rows.

File: src/Components/CreateImageWizardV2/utilities/useValidation.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import { useState } from 'react';
```
[the remark this answers is not in the document]
Comment (ezr-ondrej) [reply src/Components/CreateImageWizardV2/utilities/useValidation.tsx:1]: Right, I imlemented the pristine quite naively. Unfortunatelly I didn't find way to use the info from wizardContext in the WizardStep props (I would need it to fetch info about if the step was visited).

Given it's not easy I propose to just not have the status for now and implement it only as follow-up, let me know what you think :)

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (5.0.5) is older than required version (6.0.0)

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (5.0.5) is older than required version (6.0.0)

File: src/Components/CreateImageWizardV2/utilities/useValidation.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ezr-ondrej) [reply src/Components/CreateImageWizardV2/utilities/useValidation.tsx]: ah, I've been able to figure it out: https://github.com/osbuild/image-builder-frontend/pull/2170 but I'm not in love with it, so I'd still like it as a follow-up separate discussion :)

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (5.0.5) is older than required version (6.0.0)

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (5.0.5) is older than required version (6.0.0)

File: src/Components/CreateImageWizardV2/utilities/useValidation.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (regexowl) [reply src/Components/CreateImageWizardV2/utilities/useValidation.tsx]: I don't mind splitting the status to a follow up. Let's figure out the basic hook validation first. :+1:

**Comment by regexowl** [comment]
@ezr-ondrej Can you please resolve the conflict? I'd say we're good to merge after. The PR looks good, let's try to figure out the status in a follow up :)

**Review by regexowl — APPROVED** [review]
Thank you! :sparkles:

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (5.0.5) is older than required version (6.0.0)

**Pull request merged**